### PR TITLE
Dashboard: Use Interval variable in DashboardScene

### LINKED ISSUE #75831: @axelavargas Use Interval variable in DashboardScene

### PR DESCRIPTION
**Pull request by axelavargas**
**What is this feature?**

Implement Interval Variable in Dashboard Scene

Tasks:
- [x] Verify scenes package has the interval variable released
- [x] Check if we need to add a test 
- [x] Implement transformation from scenes to dashboard core
- [x] Implement transformation from dashboard to scenes

**Why do we need this feature?**

Feature parity 

**Which issue(s) does this PR fix?**:

Fixes #75831

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts [hunk 1]
```diff
@@ -277,6 +279,20 @@ export function createSceneVariableFromVariableModel(variable: VariableModel): S
       isMulti: variable.multi,
       hide: variable.hide,
     });
+  } else if (isIntervalVariable(variable)) {
+    return new IntervalVariable({
+      ...commonProperties,
+      value: variable.current.value,
+      text: variable.current.text,
+      description: variable.description,
+      intervals: variable.query,
```
Comment (axelavargas) [inline public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts:288]: need to split the `query` into an array

**Comment by dprokop** [comment]
@axelavargas it seems like the auto option doesn't work as expected when switching to a scene dashboard:


https://github.com/grafana/grafana/assets/2376619/d48aaaa8-75c6-47d8-9de1-b3eb236e3b41

**Comment by axelavargas** [comment]
@dprokop great catch 🙌🏾 , I added a [fix](https://github.com/grafana/grafana/pull/75836/commits/7968986cd948fb72f2d90fc25aaa206255799856) to tackle that use case :)

**Review by dprokop — APPROVED** [review]
Now it works like a charm 🥳

**Pull request merged**